ocaml-lsp-server/src/document.ml

```diff
@@ -274,9 +274,20 @@ module Merlin = struct
     ; doc : string option
     }
 
-  let type_enclosing doc pos =
+  let type_enclosing doc pos verbosity =
     with_pipeline_exn doc (fun pipeline ->
         let command = Query_protocol.Type_enclosing (None, pos, None) in
+        let pipeline =
+          match verbosity with
+          | 0 -> pipeline
+          | verbosity ->
+            let source = source doc in
+            let config = Mpipeline.final_config pipeline in
+            let config =
+              { config with query = { config.query with verbosity } }
+            in
+            Mpipeline.make config source
```

Comment:
I think this code needs to be adapted if this PR is merged after #904.

Reply:
that PR was merged, so is there a need for the code to be adapted?

Reply:
The current code looks fine to me. You aren't doing any illegal operations outside the merlin thread.